packages/eslint-plugin/docs/rules/no-duplicate-type-constituents.md

```diff
@@ -0,0 +1,64 @@
+---
+description: 'Disallow duplicate constituents of union or intersection types.'
+---
+
+> 🛑 This file is source code, not the primary documentation location! 🛑
+>
+> See **https://typescript-eslint.io/rules/no-duplicate-type-constituents** for documentation.
+
+Although TypeScript supports duplicate union and intersection constituents, people usually expect members to have unique values within the same intersection and union. Duplicate values make the code redundant and generally reduce readability.
```

Comment:
Just proofreading a bit to make it more clear what a "constituent" is for someone who hasn't heard the term before, like how https://typescript-eslint.io/rules/no-redundant-type-constituents does. How does this make you feel?

```suggestion
Although TypeScript supports types ("constituents") in union and intersection types being duplicates of each other, developers typically expect each constituent to be unique within the same intersection and union.
Duplicate values make the code overly verbose and generally reduce readability.
```

Reply:
Looks great! 
In fact, I was not familiar with the word `constituents` until I started working on this PR, but I think this sentence is much easier to understand.

Reply:
Haha yeah I learned a ton of random new words working on linting tools. It's why I recommend them to people as a great type of open source project to dive into!

This PR is shaping up nicely btw. I'm really excited to see it in. Awesome work @sajikix! 👏